proposal.rst

```diff
@@ -168,27 +171,27 @@ Project Details
 
 **Describe any existing work and concepts on which your project is based.**
 
-FILLME
+As per the current scenario, when we perform a search the total number of results are returned by Xapian::MSet::get_matches_estimated() method. The user specifies how many matching documents they're interested in. For instance, for the first page of the web search result we're interested in the top 10 results. The xapian's matcher does all the work here but in the process it often doesn't know exaclty how many total matching documents are there so it provides an estimate of the same along with the upper and lower bounds. But now due to some recent changes, the matcher now knows the first and last dcoument with which each term matches. Hence, we will be using this to calculate better estimates also by looking into how much these ranges overlap between terms.  
 
 **Do you have any preliminary findings or results which suggest that your
 approach is possible and likely to succeed?**
 
-FILLME
+The approach was suggested in the ideas page and it seems perfectly fine. And the possibility of success is very high and the things that could possible go wrong are pretty low.
 
-**What other approaches to have your considered, and why did you reject those in
+**What other approaches have you considered, and why did you reject those in
 favour of your chosen approach?**
 
-FILLME
+I was totally fine with the approach suggested in the ideas page until this question was asked. So, what I thought was to declare and initialize a new variable and keep incrementing and displaying the results as per the user requirement (an instance of 10 in the first page) but this approach is not efficient since it will work only when the searched item is single word. But when we have and array of strings containing more than a word this approach will fail and hence I rejected it. So, I will be sticking with the approach suggested by the ideas page and as elaborated in the previous to previous question.
```

Comment:
When the search query is a single word the estimated number of matches is already exact (because it's just the number of different documents which that term occurs, which we have stored for each term).